git-verify-commit: add page (#5946)
tldr-pages/tldr: pages/common/git-verify-commit.md

# git verify-commit

> Check for GPG verification of commits.

> > If GPG verification isn't present, nothing will be returned, even if extra options are chosen.

Reviewer: I think this more succinct version is better:
```suggestion
> If no commits are verified, nothing is printed.
```

Author: Completely disagree with this suggestion

Reviewer 2: Why?

Reviewer: Then how about
```suggestion
> If no commits are verified, nothing will be printed, regardless of options specified.
```

> More information: <https://git-scm.com/docs/git-verify-commit>.

> - Check git commits for a GPG signature:

Reviewer: No need to specified that they're _Git_ commits.  Also, it's properly spelled with a capital 'G' i.e. "Git".
```suggestion
- Check commits for a GPG signature:
```

> `git verify-commit {{commit_SHA}}`

Reviewer 2: Since multiple hashes can be specified:
```suggestion
`git verify-commit {{commit_SHA1}} {{commit_SHA2}}`
```

> `git verify-commit {{commit_hash}} {{commit_hash}}`

Reviewer 2: @bl-ue do you think this is fine or should we use `{{commit_hash1}} {{commit_hash2}}`? 

@CleanMachine1 the second commit hash should also be specified in the other two examples

Author: no, the user can infer that

Reviewer: @marchersimon I'd prefer numbering them each.

@CleanMachine1 is the second hash required in the other examples?  We make it a point _not_ to have the user infer much, unless it's quite obvious (and here I'd say it _isn't_.)

Reviewer 2: But the user shouldn't have to. We generally try to keep every example as independent as possible.

Author: I would feel better if it was {{optional_commit_hash}}

Reviewer 2: How about `{{commit_hashes}}`?

Author: No, infact that makes it worse since it implies that multiple hashes are required

Reviewer 2: @CleanMachine1 have you seen @bl-ue's comment [above](https://github.com/tldr-pages/tldr/pull/5946#discussion_r631241098)? I'm sure it wasn't there yesterday. Probably another GitHub bug.

Author: I didn't see it either.

Reviewer: GH doesn't automatically load new comments in thread, very unfortunately.  You **have** to reload the page.  That comment was there yesterday, maybe you guys didn't reload so never saw it? 🤔

Author: I reload every time I get a notification..

Author: if we just add {{optional_hash(s')}} that would be sufficient I feel

Reviewer: Oh, so it can be 1 or more? not restricted to 2?

Author: It can be as many as you like

Reviewer: Ah, okay.  "optional" implies _0_ or more, though, but at least one is required, correct?

Author: Yes, without 1 you get this

```
$ git verify-commit
usage: git verify-commit [-v | --verbose] <commit>...

    -v, --verbose         print commit contents
    --raw                 print raw gpg status output

```

Author: the command should be `git verify-commit {{commit_hash}} {{optional_commit_hash(s')}}

Reviewer: We actually have a special syntax just for cases like this:
```
git verify-commit {{commit_hash1 commit_hash2 ...}}
```

Author: Im going to add a commit and you tell me what you think.

Author: that implies you need more than 1

Reviewer 2: > the command should be `git verify-commit {{commit_hash}} {{optional_commit_hash(s')}}`

While technically correct, it just looks overly complicetad. Using `{{commit_hash1 commit_hash2 ...}}` is way cleaner and it's nothing tldr-specific. When you specify a gerenal list you can also write `item1, item2, item3, ...`, but everyone knows, that a list can also contain only one element.

Author: https://github.com/tldr-pages/tldr/pull/5946/commits/52a31ad6edd8b4ceb37d3b4f9ce5dc2b440c56a7 fixes the issues however makes it look a bit uglier, however it doesn't make the user take a guess at anything

Reviewer 3: We used to recommend something similar to `Check a space-separated list of commit hashes` but I think @bl-ue replaced this format with another proposal, but `optional_commit_hash(s')`is not friendly based on our format accross several pages.

Author: Give a suggestion quote and I'll approve

Reviewer 4: This conversation is on an outdated comment, which means that suggestions are disabled.

Author: Add to new

> `git verify-commit {{commit SHA hash}}`

Author: changed with https://github.com/tldr-pages/tldr/pull/5946/commits/743d7db02b40d6aa505103eab990597ab1568c34

> `git verify-commit {{commit SHA hash}}`

Reviewer 2: Why do we even mention, that it's a SHA? Wouldn't `commit_hash` be much easier?

> `git verify-commit {{commit SHA hash}}`

Author: Ok that can be done

> `git verify-commit {{commit_hash}} {{optional_commit_hash(s')}}`

Author: Add suggestions here

Reviewer: I like `{{commit_hash1 commit_hash2 ...}}`.  I don't think it indicates much that 2+ commits are required, and we do it in quite a few other places.

Reviewer: But, at least, `optional_commit_hash(s')` is wrong — it should be `optional_commit_hash(es)`, at least.

Reviewer 3: ```suggestion
`git verify-commit {{commit_hash1 commit_hash2 ...}}`
```

Author: What it is 

`git verify-commit {{commit_hash1 optional_commit_hash2 ...}}`

I think the ... is needed now, however as a user I would feel you need at least 2 to use the command

Reviewer: I like that 👍🏻

Author: should all of them have it? or just the top command?

Reviewer: Oh, I didn't see this.  There are few enough commands so that it's fine if they all have it, I think.

Author: Resolving, no one said anything about it when the wrong commands (irrelevant) were in all of the commands

> - Check commits for a GPG signature and show details of specified commit:

Reviewer 2: ```suggestion
- Check commits for a GPG signature and show details of each commit:
```

> - Check git commits for a GPG signature and show details of specified commit:

Reviewer: ```suggestion
- Check commits for a GPG signature and show details of specified commit:
```

> `git verify-commit {{commit SHA hash}} -v`

Author: I need some input from reviewers for this one, should I use the single letter command or the longer one?

> `git verify-commit {{commit SHA hash}} --verbose`

Author: changed with https://github.com/tldr-pages/tldr/pull/5946/commits/743d7db02b40d6aa505103eab990597ab1568c34

> - Check commits for a GPG signature and give RAW details:

Reviewer 2: ```suggestion
- Check commits for a GPG signature and print raw details:
```

> - Check git commits for a GPG signature and give RAW details:

Reviewer: ```suggestion
- Check commits for a GPG signature and give RAW details:
```

> - Check commits for a GPG signature and print raw details:

Reviewer 4: Perhaps this will read better?
```suggestion
- Check commits for a GPG signature and print the raw details:
```

> `git verify-commit {{commit SHA hash}} --raw`

Author: changed with https://github.com/tldr-pages/tldr/pull/5946/commits/743d7db02b40d6aa505103eab990597ab1568c34
